### ISSUE TITLE
How were the hyperparameters chosen for OGB?

### ISSUE THREAD

**Issue by Chillee** [PaddlePaddle/PGL#128]
^

You write in the paper the parameters that were selected, but could you go into more detail about how the hyperparameters were chosen? Like, what ranges were you looking at, etc.?

**Comment by sys1874** [comment]
Hi，for the OGBN，we just follow the previous studies, where the main hyperparameters we tune are as following:
lr = [0.01, 0.001]
dropout=[0, 0.1, 0.3]
The UniMP model: 
n_layer=[3, 5, 7]
n_head=[1, 2, 4]
hidden size=64, 128, 256.
We would like to make the model as large as possible in the limited GPU memory.

**Comment by Chillee** [comment]
Can you explain how they were tuned? Random search, bayesian optimization, hand tuned?

**Comment by sys1874** [comment]
In fact, we get the basic hyperparameters from other OGB baselines, and then hand tune its based on the performances from experiments. The label rate we tuned on [0.375, 0.5, 0.625]. 
So, it may not be the best hyperparameters for our model, you can perform a grid search for the hyperparameters to make a better performance.

**Comment by Chillee** [comment]
Thank you! One last question, how long did the experiments for OGB take to run?

**Comment by sys1874** [comment]
Since it needs to run 10 times for each datasets, you can get the final result for arxiv within one day,  products and proteins within several days.

**Comment by Chillee** [comment]
@sys1874 do you have exact per-epoch numbers? :) It would be helpful for my comparisons.

**Comment by sys1874** [comment]
Of course, :)

==================================================================================

1.  Product

Run: 00, Epoch: 49, Loss: 11.7688, Train: 95.34%, Valid: 92.50%, Test: 80.81% 
max_val: 92.50%, max_val_Test: 80.81%, max_val_step: 49

Run: 00, Epoch: 59, Loss: 11.5302, Train: 95.65%, Valid: 92.87%, Test: 82.48% 
max_val: 92.87%, max_val_Test: 82.48%, max_val_step: 59

Run: 00, Epoch: 69, Loss: 11.0474, Train: 95.82%, Valid: 92.99%, Test: 82.38% 
max_val: 92.99%, max_val_Test: 82.38%, max_val_step: 69

Run: 00, Epoch: 79, Loss: 10.9422, Train: 95.99%, Valid: 93.02%, Test: 82.76% 
max_val: 93.02%, max_val_Test: 82.76%, max_val_step: 79

Run: 00, Epoch: 89, Loss: 10.4045, Train: 96.00%, Valid: 93.10%, Test: 82.90% 
max_val: 93.10%, max_val_Test: 82.90%, max_val_step: 89

Run: 00, Epoch: 99, Loss: 10.2487, Train: 96.09%, Valid: 93.24%, Test: 83.13% 
max_val: 93.24%, max_val_Test: 83.13%, max_val_step: 99

==================================================================================

2.  Proteins

Run: 00, Epoch: 1800, Train: 93.43%, Valid: 91.59%, Test: 86.42% 
max_val: 91.59%, max_val_Test: 86.42%, max_val_step: 1800

Run: 00, Epoch: 1801, Train: 93.70%, Valid: 91.54%, Test: 86.29% 
max_val: 91.59%, max_val_Test: 86.42%, max_val_step: 1800

Run: 00, Epoch: 1802, Train: 93.74%, Valid: 91.61%, Test: 86.38% 
max_val: 91.61%, max_val_Test: 86.38%, max_val_step: 1802

Run: 00, Epoch: 1803, Train: 93.73%, Valid: 91.44%, Test: 86.38% 
max_val: 91.61%, max_val_Test: 86.38%, max_val_step: 1802

Run: 00, Epoch: 1804, Train: 93.53%, Valid: 91.45%, Test: 86.07% 
max_val: 91.61%, max_val_Test: 86.38%, max_val_step: 1802

Run: 00, Epoch: 1805, Train: 93.57%, Valid: 91.46%, Test: 86.32% 
max_val: 91.61%, max_val_Test: 86.38%, max_val_step: 1802
.
.
.
Run: 00, Epoch: 1995, Train: 93.33%, Valid: 91.62%, Test: 86.37% 
max_val: 91.67%, max_val_Test: 86.40%, max_val_step: 1969

Run: 00, Epoch: 1996, Train: 93.54%, Valid: 91.65%, Test: 86.26% 
max_val: 91.67%, max_val_Test: 86.40%, max_val_step: 1969

Run: 00, Epoch: 1997, Train: 93.32%, Valid: 91.63%, Test: 86.43% 
max_val: 91.67%, max_val_Test: 86.40%, max_val_step: 1969

Run: 00, Epoch: 1998, Train: 93.43%, Valid: 91.57%, Test: 86.38% 
max_val: 91.67%, max_val_Test: 86.40%, max_val_step: 1969

Run: 00, Epoch: 1999, Train: 93.41%, Valid: 91.65%, Test: 86.36% 
max_val: 91.67%, max_val_Test: 86.40%, max_val_step: 1969

==================================================================================

3.  Arxiv

Run: 00, Epoch: 99, Loss: 1.0017, Train: 79.05%, Valid: 72.02%, Test: 71.00% 
max_val: 72.14%, max_val_Test: 70.84%, max_val_step: 98

Run: 00, Epoch: 199, Loss: 0.9197, Train: 79.99%, Valid: 72.93%, Test: 71.76% 
max_val: 72.98%, max_val_Test: 71.70%, max_val_step: 190

Run: 00, Epoch: 299, Loss: 0.8737, Train: 80.29%, Valid: 73.40%, Test: 72.17% 
max_val: 73.47%, max_val_Test: 72.29%, max_val_step: 293

Run: 00, Epoch: 399, Loss: 0.8675, Train: 80.32%, Valid: 73.27%, Test: 71.43% 
max_val: 73.64%, max_val_Test: 72.29%, max_val_step: 355

Run: 00, Epoch: 499, Loss: 0.8565, Train: 80.40%, Valid: 73.58%, Test: 72.10% 
max_val: 73.74%, max_val_Test: 72.48%, max_val_step: 494
.
.
.
Run: 00, Epoch: 1499, Loss: 0.7699, Train: 81.00%, Valid: 74.16%, Test: 72.23% 
max_val: 74.63%, max_val_Test: 73.36%, max_val_step: 1342

Run: 00, Epoch: 1599, Loss: 0.7589, Train: 81.19%, Valid: 74.42%, Test: 72.54% 
max_val: 74.65%, max_val_Test: 73.08%, max_val_step: 1533

Run: 00, Epoch: 1699, Loss: 0.7572, Train: 81.23%, Valid: 74.08%, Test: 72.24% 
max_val: 74.65%, max_val_Test: 73.08%, max_val_step: 1533

Run: 00, Epoch: 1799, Loss: 0.7406, Train: 81.21%, Valid: 73.98%, Test: 71.52% 
max_val: 74.65%, max_val_Test: 73.08%, max_val_step: 1533

Run: 00, Epoch: 1899, Loss: 0.7326, Train: 81.38%, Valid: 74.31%, Test: 72.45% 
max_val: 74.65%, max_val_Test: 73.08%, max_val_step: 1533

Run: 00, Epoch: 1999, Loss: 0.7305, Train: 81.56%, Valid: 74.45%, Test: 72.87% 
max_val: 74.65%, max_val_Test: 73.08%, max_val_step: 1533

==================================================================================

4. Arxiv_large

Run: 00, Epoch: 99, Loss: 0.9357, Train: 78.96%, Valid: 72.67%, Test: 71.81% 
max_val: 72.69%, max_val_Test: 71.63%, max_val_step: 90

Run: 00, Epoch: 199, Loss: 0.8679, Train: 80.02%, Valid: 73.49%, Test: 72.58% 
max_val: 73.66%, max_val_Test: 72.77%, max_val_step: 187

Run: 00, Epoch: 299, Loss: 0.8439, Train: 80.32%, Valid: 73.07%, Test: 71.51% 
max_val: 74.02%, max_val_Test: 73.09%, max_val_step: 296

Run: 00, Epoch: 399, Loss: 0.8247, Train: 80.92%, Valid: 74.10%, Test: 73.01% 
max_val: 74.21%, max_val_Test: 73.69%, max_val_step: 394

Run: 00, Epoch: 499, Loss: 0.8151, Train: 81.22%, Valid: 73.83%, Test: 72.48% 
max_val: 74.33%, max_val_Test: 73.29%, max_val_step: 466
.
.
.
Run: 00, Epoch: 1499, Loss: 0.7061, Train: 83.26%, Valid: 73.82%, Test: 71.89% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

Run: 00, Epoch: 1599, Loss: 0.6864, Train: 83.28%, Valid: 74.22%, Test: 72.70% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

Run: 00, Epoch: 1699, Loss: 0.6747, Train: 83.56%, Valid: 74.07%, Test: 72.35% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

Run: 00, Epoch: 1799, Loss: 0.6685, Train: 83.60%, Valid: 74.05%, Test: 72.23% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

Run: 00, Epoch: 1899, Loss: 0.6516, Train: 83.54%, Valid: 73.70%, Test: 71.50% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

Run: 00, Epoch: 1999, Loss: 0.6524, Train: 83.35%, Valid: 73.99%, Test: 72.48% 
max_val: 74.64%, max_val_Test: 73.79%, max_val_step: 1004

**Comment by Chillee** [comment]
ah, I meant run-time numbers :)

**Comment by sys1874** [comment]
Sorry to hear about that, we just run this program on 32GB GPU, making full use of it. You can try to set a small size for the model to run it up.

**Comment by Chillee** [comment]
Ah, do you not have run-time numbers? Like, how many seconds each epoch takes?

If you don't have them it's perfectly understandable - it'd have to be something you're specifically logging.

**Comment by sys1874** [comment]
Maybe the Arxiv dataset we can give you run-time numbers, about 1.5 sec per epoch for UniMP， and 5 sec per epoch for UniMP_large.

**Comment by Chillee** [comment]
Haha, any guesses for products? They don't need to be perfect - I'm just trying to get a rough idea of UniMP runtime vs. other models.

**Comment by sys1874** [comment]
It may take about 1 minute per epoch for training products. and about 30 minutes for each evaluation. I guess .......

**Comment by Chillee** [comment]
Haha, thanks :)

Yeah, evaluation is ridiculously slow with these larger graphs.

**Comment by sys1874** [comment]
Haha, because there are some Bugs in our multi-process reader, we are trying to fix it up, and then it will go well in evaluation.

**Comment by Chillee** [comment]
ah, I thought that was inherent - evaluation is hideously slow for all the other cluster-GCN methods too.

**Comment by sys1874** [comment]
(https://aistudio.baidu.com/aistudio/index?lang=en) For free 32G V100, it may help you to run this program up :).
